### PR TITLE
Add substrate (optional connector) to bipolar transistor models 

### PR DESCRIPTION
**Pull request by christiankral**
Refs #2146 and  #3178. This PR does not (yet) handle #3168.

This ticket covers two issues:

1. I actually consider using only implicit grounding to be a bug. This PR now includes a conditional substrate connector. The substrate connector is not available by default for compatibility reasons.  

2. I implemented initial conditions of the PNP transistor model and consider this as enhancement. Now the NPN and PNP transistor models are build with equal features.

### CODE REVIEW

**Comment by beutlich** [comment]
@christiankral The `v4.0.0-dev` label is only meant for issues that orginate during the current development of MSL v4.0.0 **and** are not prersent in earlier MSL versions. (See the label description). This label is meant to exclude such isses from the automatically generated release notes.

File: Modelica/Electrical/Analog/Semiconductors.mo [hunk 1]
```diff
@@ -559,10 +566,13 @@ Stefan Vorkoetter - new model proposed.</li>
     Capcje = smooth(1, Cje*powlin(vbe/Phie, Me));
     cbc = smooth(1, Taur*is_t/(NR*vt_t)*exlin2(vbc/(NR*vt_t), EMin, EMax) + Capcjc);
     cbe = smooth(1, Tauf*is_t/(NF*vt_t)*exlin2(vbe/(NF*vt_t), EMin, EMax) + Capcje);
-    C.i = (ibe - ibc)*qbk - ibc/br_t - cbc*der(vbc) + Ccs*der(C.v);
+    C.i = (ibe - ibc)*qbk - ibc/br_t - cbc*der(vbc) - iS;
     B.i = ibe/bf_t + ibc/br_t + cbc*der(vbc) + cbe*der(vbe);
-    E.i = -B.i - C.i + Ccs*der(C.v);
-
+    E.i = -B.i - C.i - iS;
+    iS = -Ccs * (der(C.v) - der(vS));
```
Comment (JoeRiel) [inline Modelica/Electrical/Analog/Semiconductors.mo]: Would is = -Css*der(vcs) be clearer?

File: Modelica/Electrical/Analog/Semiconductors.mo [hunk 2]
```diff
@@ -671,10 +699,13 @@ Stefan Vorkoetter - new model proposed.</li>
     Capcje = smooth(1, Cje*powlin(veb/Phie, Me));
     ccb = smooth(1, Taur*is_t/(NR*vt_t)*exlin2(vcb/(NR*vt_t), EMin, EMax) + Capcjc);
     ceb = smooth(1, Tauf*is_t/(NF*vt_t)*exlin2(veb/(NF*vt_t), EMin, EMax) + Capcje);
-    C.i = icb/br_t + ccb*der(vcb) + Ccs*der(C.v) + (icb - ieb)*qbk;
+    C.i = icb/br_t + ccb*der(vcb) + (icb - ieb)*qbk - iS;
     B.i = -ieb/bf_t - icb/br_t - ceb*der(veb) - ccb*der(vcb);
-    E.i = -B.i - C.i + Ccs*der(C.v);
-
+    E.i = -B.i - C.i - iS;
+    iS = -Ccs * (der(C.v) - der(vS));
```
Comment (JoeRiel) [inline Modelica/Electrical/Analog/Semiconductors.mo]: Same comment as for npn, maybe iS = -Ccs * der(vcs) would be more clear.

Comment (christiankral) [reply Modelica/Electrical/Analog/Semiconductors.mo]: Yes indeed, this makes sense

**Pull request merged**